x-pack/plugins/data_visualizer/public/application/index_data_visualizer/embeddables/field_stats/field_stats_factory.tsx

```diff
@@ -0,0 +1,327 @@
+      const dataViews$ = new BehaviorSubject<DataView[] | undefined>(
+        state.dataViewId !== '' && defaultDataViewId
+          ? [await deps.data.dataViews.get(state.dataViewId ?? defaultDataViewId)]
+          : undefined
+      );
+
+      const subscriptions = new Subscription();
+      subscriptions.add(
+        fieldStatsControlsApi.dataViewId$
+          .pipe(
+            skip(1),
+            skipWhile((dataViewId) => dataViewId === '' && defaultDataViewId === null),
+            switchMap((dataViewId) => deps.data.dataViews.get(dataViewId ?? defaultDataViewId))
+          )
+          .subscribe((nextSelectedDataView) => {
+            dataViews$.next([nextSelectedDataView]);
+          })
+      );
+
+      const api = buildApi(
+        {
+          ...timeRangeApi,
+          ...titlesApi,
+          ...fieldStatsControlsApi,
+          ...dataLoadingApi,
+          getTypeDisplayName: () =>
+            i18n.translate('xpack.dataVisualizer.fieldStats.typeDisplayName', {
+              defaultMessage: 'field statistics',
+            }),
+          isEditingEnabled: () => true,
+          onEdit: async () => {
+            try {
+              const { resolveEmbeddableFieldStatsUserInput } = await import(
+                './resolve_field_stats_embeddable_input'
+              );
+              const chartState = serializeFieldStatsChartState();
+              const nextUpdate = await resolveEmbeddableFieldStatsUserInput(
+                coreStart,
+                pluginStart,
+                parentApi,
+                uuid,
+                chartState,
+                fieldStatsControlsApi
+              );
+
+              fieldStatsControlsApi.updateUserInput(nextUpdate);
+            } catch (e) {
+              return Promise.reject(e);
+            }
+          },
+          dataViews: dataViews$,
+          serializeState: () => {
+            const dataViewId = fieldStatsControlsApi.dataViewId$?.getValue();
+            const references: Reference[] = dataViewId
+              ? [
+                  {
+                    type: DATA_VIEW_SAVED_OBJECT_TYPE,
+                    name: FIELD_STATS_DATA_VIEW_REF_NAME,
+                    id: dataViewId,
+                  },
+                ]
+              : [];
+            return {
+              rawState: {
+                timeRange: undefined,
+                ...serializeTitles(),
+                ...serializeTimeRange(),
+                ...serializeFieldStatsChartState(),
+              },
+              references,
+            };
+          },
+        },
+        {
+          ...timeRangeComparators,
+          ...titleComparators,
+          ...fieldStatsControlsComparators,
+        }
+      );
+
+      const reload$ = fetch$(api).pipe(
+        skipWhile((fetchContext) => !fetchContext.isReload),
+        map(() => Date.now())
+      );
+      const query$ = fetch$(api).pipe(
```

Comment:
Is there a reason you created a new subscription for each piece of state from fetchContext instead of using the [useFetchContext](https://github.com/elastic/kibana/blob/main/packages/presentation/presentation_publishing/interfaces/fetch/fetch.ts#L152) utility method. Here is an example of where that is used [x-pack/plugins/observability_solution/infra/public/components/log_stream/log_stream_react_embeddable.tsx](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/infra/public/components/log_stream/log_stream_react_embeddable.tsx#L60)

Reply:
Was not aware for this useFetchContext utility. Thanks for the hint!

Reply:
Updated here https://github.com/elastic/kibana/pull/184030/commits/f990d2278cc309c21b0893cd076b8e2f2ec127b1